server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -8487,10 +8536,15 @@ public Boolean getDestroyRootVolumeOnVmDestruction(Long domainId){
         return DestroyRootVolumeOnVmDestruction.valueIn(domainId);
     }
 
-    private void setVncPasswordForKvmIfAvailable(Map<String, String> customParameters, UserVmVO vm){
+    private void setVncPasswordForKvmIfAvailable(Map<String, String> customParameters, UserVmVO vm) {
         if (customParameters.containsKey(VmDetailConstants.KVM_VNC_PASSWORD)
                 && StringUtils.isNotEmpty(customParameters.get(VmDetailConstants.KVM_VNC_PASSWORD))) {
             vm.setVncPassword(customParameters.get(VmDetailConstants.KVM_VNC_PASSWORD));
         }
     }
+
+    @Override
+    public Map<Long, Boolean> getDiskOfferingSuitabilityForVm(long vmId, List<Long> diskOfferingIds) {
+        return _itMgr.getDiskOfferingSuitabilityForVm(vmId, diskOfferingIds);
```

Comment:
why did we need this in both uservmmanager and virtualmachinemanager ? We can keep it in one place right!

Reply:
StoragePoolAllocators were not accessible in `UserVmManagerImpl` so it calls VitualMachineManager. I'll check using VirtualMachineManager itself in ApiDBUtils